Refactor ixblue_c3_ins package

hidmic opened the pull request:
# Description

This patch refactors the `ixblue_c3_ins` package to:

- Publish an auv_interfaces::State message (as introduced in #86)
- Gracefully terminate on error and on Ctrl+C.
- Normalize style and layout, and setup linters.

## Type of change

Please mark options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] The code builds clean without any errors or warnings
- [ ] Unit tests are passing
- [x] Linter tests are passing

# How To Test

## Test A

Launch INS driver node.

```sh
roslaunch ixblue_c3_ins ixblue_c3_ins.launch
```

Use `rostopic echo` to make sure it is publishing as expected.

## Test B

Verify `c3_ins.py` emulator-based demo is working (as described in the README.txt)

hidmic commented:
I have yet to test this on the vehicle.

patapufate commented on ixblue_c3_ins/include/ixblue_c3_ins/ixblue_c3_ins.h:
```diff
@@ -0,0 +1,45 @@
+private:
+  void publish(const c3_protocol::nav_long::nav_long_data_t& data);
+
+  ros::NodeHandle nh_;
```
Why do we have 2 nodeHandler?

hidmic replied:
A public node handle and a private node handle. The public one is at the root namespace, and it is useful for topic interfaces ('/state'). The private one uses the fully qualified node name as namespace, and it is useful for parameters (`/ins_name/some_param`).

patapufate replied:
Ok

patapufate commented on ixblue_c3_ins/CMakeLists.txt:
```diff
@@ -1,99 +1,36 @@
 ## Compile as C++11, supported in ROS Kinetic and newer
 add_compile_options(-std=c++11)
 
 ## Find catkin macros and libraries
```
nit. remove unnecessary boilerplate

patapufate commented on ixblue_c3_ins/include/ixblue_c3_ins/c3_protocol.h:
```diff
@@ -0,0 +1,78 @@
+  ~nav_long();
+  // accessors
+
+  struct nav_long_data_t
```
I see different data types between nav_long_data_t struct and NavigationLong.msg eg:
- int32_t latitude / float64 latitude

hidmic replied:
That's right ! The INS sends latitude and longitude information as signed 32 bit integers with a quantization constant (check NAVIGATION LONG packet in the [INS manual](https://www.igp.de/manuals/7-INS-InterfaceLibrary_MU-INSIII-AN-001-O.pdf)), but we publish them as floating point quantities after applying the corresponding constant.

patapufate replied:
Ok

hidmic commented:
Rebased on top #93. Added missing diagnostic checks. We'll also need to figure out how to deal with this changes in the ROS graph at the `health_monitor` level.

hidmic commented:
Rebased after merging #93.

patapufate reviewed (approved):
LGTM!

hidmic commented:
I've also tested this on the vehicle. Moving forward !
